### DISCUSSION TITLE
Artisan not recognizing Proaster

### DISCUSSION THREAD

**Discussion by gr8full21** [artisan-roaster-scope/artisan#1460] [Q&A] [answered]
Hi,  I recently purchased a Used Proaster THCR 01  roasting machine.  Can anyone help me on why Artisan software won't recognize my roaster machine? I found a old blog post about having the correct serial driver installed so that my computer will recognize the MODBUS Port. I did that but artisan still says " Modbus not Connected" .  I'm at my wits end on this one. Any help would be appreciated!!

**Comment by CarefreeBuzzBuzz** [comment]
Did you look at this page?  Does your machine have the supported temp controller?  Did you run Machine Configuration if it does?  Are you using original cables that support the data transfer?  



https://artisan-scope.org/machines/proaster/￼
Proaster
artisan-scope.org

----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Support Artisan RoasterScope  <https://artisan-scope.org/>by using 
Artisan.Plus <https://artisan.plus/en/>

**Reply by gr8full21** [reply]
Yes, I ran the machine configuration and it does have a supported temp controller.  no I'm not using the original cable.  Would that actually prevent it from transferring the data? I'll check out the link.  THANKS

**Comment by CarefreeBuzzBuzz** [comment]
Yes if the cable doesn’t support the right data transfer that could be it.  
----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Support Artisan RoasterScope  <https://artisan-scope.org/>by using 
Artisan.Plus <https://artisan.plus/en/>

**Comment by roasterdave** [comment]
MODBUS errors are usually due to configuration mismatches.  In rarer cases the hardware is faulty.  As already mentioned, be sure the USB cable is of good quality.  For more helpful answers you need to tell us what kind of computer and operating system version, what version of Artisan and what serial driver you installed?  Did the previous owner have the roaster connected to any logging software?

First, check that the Comm Port set in the MODBUS config is correct (Config>> Ports>> tab-Modbus).  How you check which comm port is used for the roaster's USB varies by operating system, you may need Google's help. 

Next be sure the baud rate is correct.  Since your roaster was preowned some of the factory settings may have been changed.  You can experiment with the baud rate.  Try different baud rates to see if any allow it to connect.  Since you used Config>> Machine>> Proaster, the other serial settings should be correct.  Let us know if you get it connected or not.

**Comment by MAKOMO** [comment, marked as answer]
Are you sure the serial port of your roaster is recognized? You can simply check by opening the ports popup in the MODBUS tab while the roaster is connected and again while it is not connected via USB. The one additional entry that pops up while the roaster is connected is the one for your roasting machine (or rather its USB2485 converter (RS485 is the link used by serial MODBUS RTU). 

If you do not see an extra port in the test above, your computer is NOT recognizing your USB2485 converter, most likely as the correct driver (for its chipset) is not installed.

Depending on your USB2485 converter between your laptop and the machine you might need to [install a serial driver](https://artisan-scope.org/modbus_serial/). Note that the FTDI driver does not need any installation on current operating systems. In this case just install the CP210x driver and conduct the test above again. Installing a driver not used should not harm (just never install an extra FTDI driver on macOS).

**Reply by gr8full21** [reply]
thank you so much for your help!  This information solved my problem! I had worked at this for days.  Thanks again.
